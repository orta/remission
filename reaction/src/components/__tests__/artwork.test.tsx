import * as React from "react"
import * as renderer from "react-test-renderer"

import "jest-styled-components"

import Artwork from "../artwork"

describe("Artwork", () => {
  it("renders correctly", () => {
    const artworkProps = {
      id: "mikael-olson-some-kind-of-dinosaur",
      title: "Some Kind of Dinosaur",
      date: "2015",
      sale_message: "$875",
      is_in_auction: false,
      image: {
        placeholder: 200,
        url: "artsy.net/image-url",
        aspect_ratio: 0.74,
      },
      artists: [
        {
          __id: "mikael-olson",
          name: "Mikael Olson",
        },
      ],
      partner: {
        name: "Gallery 1261",
      },
      href: "/artwork/mikael-olson-some-kind-of-dinosaur",
    }

    const artwork = renderer.create(<Artwork artwork={artworkProps} />).toJSON()
    expect(artwork).toMatchSnapshot()
  })
})
