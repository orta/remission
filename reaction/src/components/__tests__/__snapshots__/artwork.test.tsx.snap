// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork renders correctly 1`] = `
.c5 {
  color: #333333;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  display: block;
  -webkit-text-overflow: ellipsis;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c3 {
  font-family: 'Adobe Garamond W08',  'adobe-garamond-pro',  'AGaramondPro-Regular',  'Times New Roman',  'Times',  'serif';
  -webkit-font-smoothing: antialiased;
  color: #666666;
  margin-top: 12px;
  font-size: 15px;
  -webkit-text-align: left;
  text-align: left;
  line-height: 20px;
}

.c0 {
  width: 100%;
}

.c1 {
  width: 100%;
  position: relative;
  cursor: pointer;
}

.c1:before {
  content: '';
  display: block;
  padding-bottom: 100%;
}

.c1 .overlay-container {
  -webkit-transition: opacity 0.25s;
  transition: opacity 0.25s;
}

.c1 .overlay-container.hovered {
  opacity: 0;
  visibility: hidden;
}

.c1 .overlay-container.selected {
  opacity: 1;
  visibility: visible;
}

.c2 {
  max-width: 100%;
  max-height: 100%;
  margin: auto;
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  right: 0;
}

@media (min-width: 49px) {
  .c1:hover .overlay-container.hovered {
    opacity: 1;
    visibility: visible;
  }
}

<div
  className="c0"
  onClick={[Function]}
  size={250}
>
  <div
    className="c1"
    size={250}
  >
    <img
      className="c2"
      size={250}
      src="artsy.net/image-url"
    />
    <div
      className="overlay-container"
    />
  </div>
  <div
    className="c3"
  >
    <div>
      <div
        className="c4"
      >
        <strong>
          <a
            className="c5"
            href={undefined}
          >
            Mikael Olson
          </a>
        </strong>
      </div>
      <div
        className="c4"
      >
        <a
          className="c5"
          href="/artwork/mikael-olson-some-kind-of-dinosaur"
        >
          <em>
            Some Kind of Dinosaur
          </em>
          , 2015
        </a>
      </div>
      <div
        className="c4"
      >
        <a
          className="c5"
          href={undefined}
        >
          Gallery 1261
        </a>
      </div>
    </div>
  </div>
</div>
`;
